lib/rollbax.ex

```diff
@@ -57,28 +74,51 @@ defmodule Rollbax do
 
   @doc false
   def start(_type, _args) do
-    enabled = Application.get_env(:rollbax, :enabled, true)
-    custom = Application.get_env(:rollbax, :custom, %{})
-    api_endpoint = Application.get_env(:rollbax, :api_endpoint, @default_api_endpoint)
-    environment = resolve_system_env(Application.fetch_env!(:rollbax, :environment))
-
-    access_token =
-      case enabled do
-        true -> resolve_system_env(Application.fetch_env!(:rollbax, :access_token))
-        _other -> :not_needed
-      end
+    config = read_config()
 
-    if Application.get_env(:rollbax, :crash_reports, [])[:enabled] do
+    if config[:enabled] == true and is_nil(config[:access_token]) do
+      raise ArgumentError, ":access_token is required when :enabled is true"
+    end
+
+    if config[:enable_crash_reports] do
+      # We do this because the handler will read `:repoters` out of the app's environment.
+      Application.put_env(:rollbax, :reporters, config[:reporters])
       :error_logger.add_report_handler(Rollbax.Logger)
     end
 
     children = [
-      Supervisor.Spec.worker(Rollbax.Client, [api_endpoint, access_token, environment, enabled, custom])
+      Supervisor.Spec.worker(Rollbax.Client, [config])
     ]
 
     Supervisor.start_link(children, strategy: :one_for_one)
   end
 
+  defp read_config() do
+    environment = Application.get_env(:rollbax, :environment)
+    access_token = Application.get_env(:rollbax, :access_token)
+
+    config = [
+      enabled: Application.get_env(:rollbax, :enabled, true),
+      custom: Application.get_env(:rollbax, :custom, %{}),
+      api_endpoint: Application.get_env(:rollbax, :api_endpoint, @default_api_endpoint),
+      enable_crash_reports: Application.get_env(:rollbax, :crash_reports, []),
+      reporters: Application.get_env(:rollbax, :reporters, [Rollbax.Reporter.Standard])
+    ]
```

Comment:
It would be more idiomatic to set these default values in the `env` entry in mix.exs file and call `Application.get_all_env(:rollbax)`.

Reply:
Good catch @fishcakez, thanks <3